File: laba1.cpp
#include <iostream>
#include <string>
#include <stdexcept>
#include <iomanip>

class InternetShop {
private:
    std::string shopName;
    double tvPrice;
    int purchaseCount;

public:
    InternetShop(const std::string& name, double price, int count) {
        setShopName(name);
        setTvPrice(price);
        setPurchaseCount(count);
    }

    void setShopName(const std::string& name) {
        if (name.empty()) {
            throw std::invalid_argument("У магазина должно быть название");
        }
        shopName = name;
    }

    void setTvPrice(double price) {
        if (price <= 0) {
            throw std::invalid_argument("Цена телевизора меньше 0, повышай");
        }
        if (price > 300000) {
            throw std::invalid_argument("Цена телевизора слишком дорогая, понижай");
        }
        tvPrice = price;
    }

    void setPurchaseCount(int count) {
        if (count < 0) {
            throw std::invalid_argument("Товары не могут быть 0 или меньше 0");
        }
        if (count > 100) {
            throw std::invalid_argument("Товаров может быть только 100, не больше");
        }
        purchaseCount = count;
    }

    std::string getShopName() const { return shopName; }
    double getTvPrice() const { return tvPrice; }
    int getPurchaseCount() const { return purchaseCount; }

    double calcRevenue() const {
        return tvPrice * purchaseCount;
    }

    void setRevenueOnly(double revenue) {
        if (revenue < 0) {
            throw std::invalid_argument("Меньше нуля нельзя заработать с продажи");
        }
    }
};

int main() {
    setlocale(LC_ALL, "Russian");

    try {
        InternetShop shop("PoliTechMarket", 300000, 100);

        std::cout << "Магазин: " << shop.getShopName() << std::endl;
        std::cout << "Цена телевизора: " << std::fixed << std::setprecision(0)
            << shop.getTvPrice() << " руб." << std::endl;
        std::cout << "Количество покупок: " << shop.getPurchaseCount() << std::endl;
        std::cout << "Общая выручка: " << std::fixed << std::setprecision(0)
            << shop.calcRevenue() << " руб." << std::endl;

        shop.setRevenueOnly(300000);


    }
    catch (const std::exception& e) {
        std::cerr << "Одна ошибка, и ты ошибся\n" << e.what() << std::endl;
    }

    return 0;
}

File: laba1.py
class InternetShop:
    def __init__(self, name: str, price: float, count: int):
        self.set_shop_name(name)
        self.set_tv_price(price)
        self.set_purchase_count(count)

    def set_shop_name(self, name: str):
        if not name.strip():
            raise ValueError("У магазина должно быть название")
        self._shop_name = name

    def set_tv_price(self, price: float):
        if price <= 0:
            raise ValueError("Цена телевизора меньше 0, повышай")
        if price > 300_000:
            raise ValueError("Цена телевизора слишком дорогая, понижай")
        self._tv_price = price

    def set_purchase_count(self, count: int):
        if count < 0:
            raise ValueError("Товары не могут быть 0 или меньше 0")
        if count > 100:
            raise ValueError("Товаров может быть только 100, не больше")
        self._purchase_count = count

    def get_shop_name(self) -> str:
        return self._shop_name

    def get_tv_price(self) -> float:
        return self._tv_price

    def get_purchase_count(self) -> int:
        return self._purchase_count

    def calc_revenue(self) -> float:
        return self._tv_price * self._purchase_count

    def set_revenue_only(self, revenue: float):
        if revenue < 0:
            raise ValueError("Меньше нуля нельзя заработать с продажи")
            
def main():
    try:
        shop = InternetShop("PoliTechMarket", 300_000, 100)

        print(f"Магазин: {shop.get_shop_name()}")
        print(f"Цена телевизора: {shop.get_tv_price():.0f} руб.")
        print(f"Количество покупок: {shop.get_purchase_count()}")
        print(f"Общая выручка: {shop.calc_revenue():.0f} руб.")

        shop.set_revenue_only(300_000)

    except ValueError as e:
        print("Одна ошибка, и ты ошибся")
        print(e)


if __name__ == "__main__":
    main()
